model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/InfinispanUserSessionProviderFactory.java

```diff
@@ -112,6 +112,13 @@ public void init(Config.Scope config) {
 
     @Override
     public void postInit(final KeycloakSessionFactory factory) {
+        KeycloakModelUtils.runJobInTransaction(factory, (KeycloakSession session) -> {
+            keyGenerator = new InfinispanKeyGenerator();
+            this.remoteCacheInvoker = new RemoteCacheInvoker();
+
+            // Initialize persister for periodically doing bulk DB updates of lastSessionRefresh timestamps of refreshed sessions
+            persisterLastSessionRefreshStore = new PersisterLastSessionRefreshStoreFactory().createAndInit(session, true);
```

Comment:
This may become an issue with half-initialized factories: in `postInit`, all factories have been _loaded_ but not all factories have been also fully initialized. Hence if a factory is used from the timer handler set in `createAndInit` method, it might be half-initialized and behave weirdly.
I recall there was some problem with this piece of code in the original place. Could you please remind me of what this problem exactly was?